src/access/RequestPath.h

```diff
@@ -19,15 +19,29 @@
 #pragma once
 
 #include <lib/core/DataModelTypes.h>
+#include <optional>
 
 namespace chip {
 namespace Access {
 
+enum class RequestType : uint8_t
+{
+    kRequestTypeUnknown,
+    kReadRequest,
+    kWriteRequest,
+    kInvokeRequest,
+    kSubscribeEventRequest
```

Comment:
Why is subscribe event called out here? Subscribe privileges are the same as read privileges

Reply:
In the spec:

if arl_restriction.Type == AccessRestrictionTypeEnum.EventForbidden and
action.Type == Action.TypeEnum.SubscribeRequest:
# if the event id is NULL (wildcard) or matches the cluster_element
if arl_restriction.ID == NULL or arl_restriction.ID == cluster_element:
# then no granted privileges
return set()

Maybe this needs to be clarified.  What we are really after is restricting access to an event(s), regardless of if its an attempt to read or subscribe.  These enum entries capture the requested action, which could be read or subscribe.

Reply:
Fundamentally: the spec is broken here.

Reply:
The fact that it's events separate from attributes is weird. Let's fix follow-up, though.

Reply:
Resolving conversation, but tracking here for post 1.4 continuation: https://github.com/project-chip/connectedhomeip/issues/34913